src/main/java/cuchaz/enigma/gui/GuiController.java

```diff
@@ -426,20 +424,105 @@ private void remapSource(Translator translator) {
 			return;
 		}
 
-		currentSource.remapSource(deobfuscator, translator);
+		currentSource.remapSource(project, translator);
 
 		gui.setEditorTheme(Config.getInstance().lookAndFeel);
 		gui.setSource(currentSource);
 	}
 
-	public Deobfuscator getDeobfuscator() {
-		return deobfuscator;
-	}
-
 	public void modifierChange(ItemEvent event) {
 		if (event.getStateChange() == ItemEvent.SELECTED) {
-			deobfuscator.changeModifier(gui.cursorReference.entry, (AccessModifier) event.getItem());
+			EntryRemapper mapper = project.getMapper();
+			Entry<?> entry = gui.cursorReference.entry;
+			AccessModifier modifier = (AccessModifier) event.getItem();
+
+			EntryMapping mapping = mapper.getDeobfMapping(entry);
+			if (mapping != null) {
+				mapper.mapFromObf(entry, new EntryMapping(mapping.getTargetName(), modifier));
+			} else {
+				mapper.mapFromObf(entry, new EntryMapping(entry.getName(), modifier));
+			}
+
 			refreshCurrentClass();
 		}
 	}
+
+	public ClassInheritanceTreeNode getClassInheritance(ClassEntry entry) {
```

Comment:
Apologies for the moves around here. Was going to rework this further but the PR was already going too far out of scope :P